### PR TITLE
Icon migration

### PR DESCRIPTION
**Pull request by qroll**
**Changes**
- Migrate icons to new icon library
- Standardise `SmartAppBanner` and `FeedbackRating` star icons
- Enhance `IconButton` with commonly required style options
- Add deprecation notice for `Icon` component
- [delete] branch

**Changelog entry**
- Migrate to icons from `@lifesg/react-icons`
- Deprecate `Icon`

**Additional notes**
- Might be easier to go through the commits instead of viewing the changes as a whole

### CODE REVIEW

File: src/accordion/accordion-item.style.tsx [hunk 1]
```diff
@@ -54,25 +54,18 @@ export const TitleH4 = styled(Text.H4)<StyleProps>`
 `;
 
 export const ExpandCollapseButton = styled(IconButton)<StyleProps>`
-    :active,
```
Comment (qroll) [inline src/accordion/accordion-item.style.tsx:57]: removed as it can now be styled from icon button props

File: src/alert/alert.style.tsx [hunk 2]
```diff
@@ -76,15 +76,10 @@ export const ActionLinkText = styled(Text.Hyperlink.Small)`
     margin-top: 0.5rem;
     display: flex;
     align-items: center;
-
```
Comment (qroll) [inline src/alert/alert.style.tsx:79]: removed as icon already inherits the text color

File: src/breadcrumb/breadcrumb.style.tsx [hunk 3]
```diff
@@ -109,9 +105,9 @@ export const Item = styled.li<ItemStyleProps>`
     }};
 `;
 
-export const Caret = styled(Icon)<CaretProps>`
+export const Caret = styled(ChevronRightIcon)`
     margin: 0.25rem 0.25rem 0.1rem 0.25rem;
-    color: ${(props) => props.$color || Color.Neutral[4]};
```
Comment (qroll) [inline src/breadcrumb/breadcrumb.style.tsx:114]: prop was never passed down

File: src/form/form-label-addon.style.tsx [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
+import { Color } from "../color";
 
 export const TriggerArea = styled.button`
     padding: 0.25rem 0.5rem 0.25rem 0.25rem;
     border: none;
     background: none;
     cursor: pointer;
 
-    svg {
+    & > svg {
```
Comment (qroll) [inline src/form/form-label-addon.style.tsx:10]: added default styling for form label icon to make it easier to switch out the rendered icon

File: src/icon-button/icon-button.style.tsx [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
+    $highlight: boolean;
+}
+
+export const Main = styled.button<StyleProps>`
     display: flex;
     justify-content: center;
     align-items: center;
     padding: 1.5rem;
-    background-color: ${Color.Neutral[8]};
+    background-color: transparent;
```
Comment (qroll) [inline src/icon-button/icon-button.style.tsx:14]: almost every component was overriding `background-color` from white to transparent, so set transparent as the default

File: src/shared/dropdown-list/dropdown-search.styles.tsx [hunk 6]
```diff
@@ -15,7 +17,7 @@ export const SearchInput = styled.input`
     height: 3rem;
     border: none;
     background: transparent;
-    width: 100%;
+    flex: 1;
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-search.styles.tsx:20]: to prevent `MagnifierIcon` from changing sizes when text is cleared/entered

**Review by keithtxw — APPROVED** [review]

**Pull request merged**